src/VisualStudio/Core/Def/SampleToolWindow/DocSymbol.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.ObjectModel;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    using SymbolKind = LanguageServer.Protocol.SymbolKind;
+
+    internal class DocSymbol
+    {
+        public string Name { get; set; }
+
+        public ObservableCollection<DocSymbol> Children { get; set; }
+
+        public int StartLine { get; set; }
+        public int StartChar { get; set; }
+        public int EndLine { get; set; }
+        public int EndChar { get; set; }
+
+        public SymbolKind SymbolKind { get; set; }
+        public ImageMoniker ImgMoniker { get; set; }
+
+        public bool IsExpanded { get; set; }
+        public bool IsSelected { get; set; }
+
+        public DocSymbol(string name, SymbolKind symbolKind, int startLine, int startChar, int endLine, int endChar)
```

Comment:
couldn't we just pass the `DocumentSymbol` into the constructor and assign the relevant properties in this constructor body?